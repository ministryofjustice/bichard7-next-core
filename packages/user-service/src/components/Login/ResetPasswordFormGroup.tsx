import PasswordInput from "./PasswordInput"
import React from "react"

interface ResetPasswordFormGroupProps {
  passwordMismatch: boolean | undefined
  passwordsMismatchError: string
  newPasswordError: string | false | undefined
}

const ResetPasswordFormGroup = ({
  passwordMismatch,
  passwordsMismatchError,
  newPasswordError
}: ResetPasswordFormGroupProps) => {
  const formGroupClasses = `govuk-form-group ${passwordMismatch ? "govuk-form-group--error" : ""}`

  const textInputWidth = "30"

  return (
    <div className={formGroupClasses}>
      {passwordMismatch && (
        <span id={"password-mismatch-error"} className="govuk-error-message">
          <span className="govuk-visually-hidden">{"Error:"}</span> {passwordsMismatchError}
        </span>
      )}

      <PasswordInput
        id="newPassword"
        name="newPassword"
        label="New password"
        labelSize="s"
        hint="Enter your new password"
        error={newPasswordError}
        width={textInputWidth}
      />

      <PasswordInput
        id="confirmPassword"
        name="confirmPassword"
        label="Confirm password"
        labelSize="s"
        hint="Re-type your password"
        width={textInputWidth}
      />
    </div>
  )
}

export default ResetPasswordFormGroup
