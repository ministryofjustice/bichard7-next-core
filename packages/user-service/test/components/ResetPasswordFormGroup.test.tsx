import { render } from "@testing-library/react"
import ResetPasswordFormGroup from "components/Login/ResetPasswordFormGroup"

describe("ResetPasswordFormGroup", () => {
  it("should render correctly without any errors", () => {
    const { container } = render(
      <ResetPasswordFormGroup
        passwordMismatch={false}
        passwordsMismatchError={undefined}
        newPasswordError={undefined}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should render with group error styling and message when passwordMismatch is true", () => {
    const mismatchMessage = "Passwords do not match"
    const { container } = render(
      <ResetPasswordFormGroup
        passwordMismatch={true}
        passwordsMismatchError={mismatchMessage}
        newPasswordError={undefined}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should pass newPasswordError to the specific New Password input", () => {
    const individualErrorMessage = "Your password is too short"
    const { container } = render(
      <ResetPasswordFormGroup
        passwordMismatch={false}
        passwordsMismatchError={"Should not show"}
        newPasswordError={individualErrorMessage}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
