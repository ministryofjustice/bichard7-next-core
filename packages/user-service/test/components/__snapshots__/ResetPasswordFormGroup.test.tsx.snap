// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ResetPasswordFormGroup should pass newPasswordError to the specific New Password input 1`] = `
<div>
  <div
    class="govuk-form-group "
  >
    <div
      class="govuk-form-group govuk-password-input govuk-form-group--error"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="newPassword"
      >
        New password
      </label>
      <div
        class="govuk-hint"
        id="newPassword-hint"
      >
        Enter your new password
      </div>
      <span
        class="govuk-error-message"
        data-test="password-input_newPassword-error"
        id="newPassword-error"
      >
        <span
          class="govuk-visually-hidden"
        >
          Error:
        </span>
         
        Your password is too short
      </span>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="newPassword-hint newPassword-error"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30 govuk-input--error"
          data-test="password-input_newPassword"
          id="newPassword"
          name="newPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="newPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
    <div
      class="govuk-form-group govuk-password-input"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="confirmPassword"
      >
        Confirm password
      </label>
      <div
        class="govuk-hint"
        id="confirmPassword-hint"
      >
        Re-type your password
      </div>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="confirmPassword-hint"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30"
          data-test="password-input_confirmPassword"
          id="confirmPassword"
          name="confirmPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="confirmPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResetPasswordFormGroup should render correctly without any errors 1`] = `
<div>
  <div
    class="govuk-form-group "
  >
    <div
      class="govuk-form-group govuk-password-input"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="newPassword"
      >
        New password
      </label>
      <div
        class="govuk-hint"
        id="newPassword-hint"
      >
        Enter your new password
      </div>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="newPassword-hint"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30"
          data-test="password-input_newPassword"
          id="newPassword"
          name="newPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="newPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
    <div
      class="govuk-form-group govuk-password-input"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="confirmPassword"
      >
        Confirm password
      </label>
      <div
        class="govuk-hint"
        id="confirmPassword-hint"
      >
        Re-type your password
      </div>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="confirmPassword-hint"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30"
          data-test="password-input_confirmPassword"
          id="confirmPassword"
          name="confirmPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="confirmPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ResetPasswordFormGroup should render with group error styling and message when passwordMismatch is true 1`] = `
<div>
  <div
    class="govuk-form-group govuk-form-group--error"
  >
    <span
      class="govuk-error-message"
      id="password-mismatch-error"
    >
      <span
        class="govuk-visually-hidden"
      >
        Error:
      </span>
       
      Passwords do not match
    </span>
    <div
      class="govuk-form-group govuk-password-input"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="newPassword"
      >
        New password
      </label>
      <div
        class="govuk-hint"
        id="newPassword-hint"
      >
        Enter your new password
      </div>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="newPassword-hint"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30"
          data-test="password-input_newPassword"
          id="newPassword"
          name="newPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="newPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
    <div
      class="govuk-form-group govuk-password-input"
      data-module="govuk-password-input"
    >
      <label
        class="govuk-label govuk-label--s"
        for="confirmPassword"
      >
        Confirm password
      </label>
      <div
        class="govuk-hint"
        id="confirmPassword-hint"
      >
        Re-type your password
      </div>
      <div
        class="govuk-input__wrapper govuk-password-input__wrapper"
      >
        <input
          aria-describedby="confirmPassword-hint"
          autocapitalize="none"
          autocomplete="current-password"
          class="govuk-input govuk-password-input__input govuk-js-password-input-input govuk-input--width-30"
          data-test="password-input_confirmPassword"
          id="confirmPassword"
          name="confirmPassword"
          spellcheck="false"
          type="password"
        />
        <button
          aria-controls="confirmPassword"
          aria-label="Show password"
          class="govuk-button govuk-button--secondary govuk-password-input__toggle govuk-js-password-input-toggle"
          data-module="govuk-button"
          type="button"
        >
          Show
        </button>
      </div>
    </div>
  </div>
</div>
`;
